yt/frontends/nc4_cm1/data_structures.py

```diff
@@ -0,0 +1,233 @@
+        # float type for the simulation edges and must be float64 now
+        self.float_type = np.float64
+        super(CM1Hierarchy, self).__init__(ds, dataset_type)
+
+    def _initialize_state_variables(self):
+        super(CM1Hierarchy, self)._initialize_state_variables()
+        self.num_grids = 1
+
+    def _detect_output_fields(self):
+        # build list of on-disk fields for dataset_type 'cm1'
+        self.field_list = []
+        for vname in self.dataset.parameters["variable_names"]:
+            self.field_list.append(("cm1", vname))
+
+    def _count_grids(self):
+        # This needs to set self.num_grids
+        self.num_grids = 1
+
+    def _parse_index(self):
+        # This needs to fill the following arrays, where N is self.num_grids:
+        #   self.grid_left_edge         (N, 3) <= float64
+        #   self.grid_right_edge        (N, 3) <= float64
+        #   self.grid_dimensions        (N, 3) <= int
+        #   self.grid_particle_count    (N, 1) <= int
+        #   self.grid_levels            (N, 1) <= int
+        #   self.grids                  (N, 1) <= grid objects
+        #   self.max_level = self.grid_levels.max()
+        self.grid_left_edge[0][:] = self.ds.domain_left_edge[:]
+        self.grid_right_edge[0][:] = self.ds.domain_right_edge[:]
+        self.grid_dimensions[0][:] = self.ds.domain_dimensions[:]
+        self.grid_particle_count[0][0] = 0
+        self.grid_levels[0][0] = 1
+        self.max_level = 1
+
+    def _populate_grid_objects(self):
+        # For each grid g, this must call:
+        #   g._prepare_grid()
+        #   g._setup_dx()
+        # This must also set:
+        #   g.Children <= list of child grids
+        #   g.Parent   <= parent grid
+        # This is handled by the frontend because often the children must be
+        # identified.
+        self.grids = np.empty(self.num_grids, dtype="object")
+        for i in range(self.num_grids):
+            g = self.grid(i, self, self.grid_levels.flat[i], self.grid_dimensions[i])
+            g._prepare_grid()
+            g._setup_dx()
+            self.grids[i] = g
+
+
+class CM1Dataset(Dataset):
+    _index_class = CM1Hierarchy
+    _field_info_class = CM1FieldInfo
+
+    def __init__(
+        self, filename, dataset_type="cm1", storage_filename=None, units_override=None
+    ):
+        self.fluid_types += ("cm1",)
+        self._handle = NetCDF4FileHandler(filename)
+        # refinement factor between a grid and its subgrid
+        self.refine_by = 1
+        super(CM1Dataset, self).__init__(
+            filename, dataset_type, units_override=units_override, unit_system="mks"
+        )
+        self.storage_filename = storage_filename
+        self.filename = filename
+
+    def _setup_coordinate_handler(self):
+        super(CM1Dataset, self)._setup_coordinate_handler()
+        self.coordinates._x_pairs = (("x", "y"), ("y", "x"), ("z", "x"))
+        self.coordinates._y_pairs = (("x", "z"), ("y", "z"), ("z", "y"))
```

Comment:
I believe this sets the axis ordering on slice plots so that when slicing `z` is always on the vertical axis -- @matthewturk or @keltonhalbert should confirm...

Reply:
You're correct @chrishavlin. This correctly orders the axes so plots aren't rotated.

Reply:
ok, cool. we should add a comment to the code here I think.

Reply:
Ok thank you both for clarifying this. I suggest to add an inline comment.